src/mbed_tools/build/_internal/config/source.py

```diff
@@ -78,6 +83,30 @@ def __post_init__(self) -> None:
         self.value = _sanitise_value(self.value)
 
 
+@dataclass
+class Memory:
+    """Representation of a defined RAM/ROM region."""
+
+    name: str
+    namespace: str
+    start: str
+    size: str
+
+    def __post_init__(self) -> None:
+        """Convert start and size to hex format strings."""
+        self.start = self._as_hex(self.start, "START")
+        self.size = self._as_hex(self.size, "SIZE")
+
+    def _as_hex(self, value: str, attr: str) -> str:
```

Comment:
We only use `self` for adding stuff to the logging message in this method. This seems like a weak reason to add another method to this class. I'd prefer a free-function here and possibly moving the error handling up into `__post_init__`